providers/oidc.go

```diff
@@ -207,13 +212,18 @@ func (p *OIDCProvider) createSessionStateInternal(ctx context.Context, rawIDToke
 	newSession.Email = claims.UserID // TODO Rename SessionState.Email to .UserID in the near future
 
 	newSession.User = claims.Subject
+	newSession.Groups = p.HeaderEncoder.EncodeArray(claims.Groups)
```

Comment:
This seems too early to be encoding. I think we should leave it as a `[]string` in the session object (and its encoded JSON and future msgpack form) and only do the array separater encoding at the last moment when we need it as a single string in the HTTP request header.

This way it will be easier for future code to interact with a session's Groups.

Reply:
This is done, much simpler.